### PR TITLE
Jenkins service

### PR DESCRIPTION
**Pull request by ItsAlxl**

### CODE REVIEW

File: Jenkins/jenkins.yaml [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    triggerDownstreamUponResultUnstable: false
+
+credentials:
+  system:
+    domainCredentials:
+    - credentials:
+      - string:
+          id: "github"
+          scope: GLOBAL
+          secret: "{AQAAABAAAAAwUfPZgEsGqs9O7PMnauq0dP8aN5A9rQqRXSp/nzNPKmJUIHllESOgQZ2dL9aeTzZ2lvC7P/zgao0/zWixpeJE4Q==}"
```
Comment (Bensonre) [inline Jenkins/jenkins.yaml:144]: No secrets should be present in src-ctrl

File: Jenkins/BuildAndPush_Jenkins.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+echo "Enter Docker Hub username"
+read username
+docker build -t $username/autoci-jenkins:latest $(dirname $0)
+docker push $username/autoci-jenkins:latest
```
Comment (re-blank) [inline Jenkins/BuildAndPush_Jenkins.sh:4]: Should migrate to ECR if we have time.

File: JenkinsService/src/main/java/com/revature/JenkinsServlet.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            }
+            log.error("JSON exception occurring with this JSON: " + err);
+            throw new IOException(err);
+        }
+    }
+	
+    private void makeJob(JSONObject response) throws IOException {
+        log.info("Beginning makeJob method");
+        CommandExecutor exec = new CommandExecutor();
+        String cmd = "curl -X POST -u " + jenkinsAuth + " " + jenkinsUrl + "/job/seed/buildWithParameters --data githubURL=" + repoUrl + " --data projectName=" + projName + " --data slackChannel=" + slackChannel;
```
Comment (Bensonre) [inline JenkinsService/src/main/java/com/revature/JenkinsServlet.java:97]: This isn't wrong, but why are we using curl and not writing java code to do this?

File: Jenkins/jenkins.yaml [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+  - envVars:
+      env:
+      - key: "ContainerRegistryCredId"
+        value: "ecr:us-east-1:mehrab-trg00000835-opsusr"
+      - key: "ContainerRegistryURL"
+        value: "https://855430746673.dkr.ecr.us-east-1.amazonaws.com"
+      - key: "kubectlCreds"
+        value: "kubectl-token"
+      - key: "kubectlServer"
+        value: "https://kubernetes.default.svc"
```
Comment (re-blank) [inline Jenkins/jenkins.yaml]: Add helmChartRepoURL to env if there's time: http://a9c2594943a63472faa675f1858e4264-18446664.us-east-1.elb.amazonaws.com/api/charts

File: JenkinsService/src/main/java/com/revature/JenkinsServlet.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        log.info("Beginning createContainerRepo method");
+        CommandExecutor exec = new CommandExecutor(getServletContext().getRealPath("/terra/"));
+
+        // Copy template.txt to exec.tf, populate its fields
+        Path templateFilePath = Path.of(getServletContext().getRealPath("/terra/template.txt"));
+        Path execFilePath = Path.of(getServletContext().getRealPath("/terra/temp_exec.tf"));
+
+        System.out.println("Writing temp file...");
+        String templateText = Files.readString(templateFilePath);
+        String execText = String.format(templateText, projName);
```
Comment (Bensonre) [inline JenkinsService/src/main/java/com/revature/JenkinsServlet.java:118]: Clever use of string.format()

File: JenkinsService/src/test/java/com/revature/AppTest.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+/**
+ * Unit test for simple App.
+ */
+public class AppTest 
+{
+    /**
```
Comment (Bensonre) [inline JenkinsService/src/test/java/com/revature/AppTest.java:12]: No unit testing?

File: Jenkins/jenkins.yaml [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+              id: "14.11.0"
+              npmPackagesRefreshHours: 72
+  pipelineMaven:
+    triggerDownstreamUponResultAborted: false
+    triggerDownstreamUponResultFailure: false
+    triggerDownstreamUponResultNotBuilt: false
+    triggerDownstreamUponResultSuccess: true
+    triggerDownstreamUponResultUnstable: false
+
+credentials:
```
Comment (re-blank) [inline Jenkins/jenkins.yaml:137]: We could inject these through secrets and reference them via environment variables

**Review by Bensonre — CHANGES_REQUESTED** [review]

File: Jenkins/jenkins.yaml (hunk 1 above)
Comment (Bensonre) [reply Jenkins/jenkins.yaml:144]: nvm

**Review by Bensonre — APPROVED** [review]

**Pull request merged**